menuinst/platforms/win.py

```diff
@@ -36,8 +36,14 @@ def create(self) -> Tuple[os.PathLike]:
         return (self.start_menu_location,)
 
     def remove(self) -> Tuple[os.PathLike]:
-        log.debug("Removing %s", self.start_menu_location)
-        shutil.rmtree(self.start_menu_location, ignore_errors=True)
+        # Only remove if the Start Menu directory is empty in case applications share a folder.
+        try:
+            menu_location = Path(self.start_menu_location)
+            if menu_location.exists():
+                next(menu_location.iterdir())
+        except StopIteration:
+            log.debug("Removing %s", self.start_menu_location)
+            shutil.rmtree(self.start_menu_location, ignore_errors=True)
```

Comment:
```suggestion
        menu_location = Path(self.start_menu_location)
        if menu_location.exists():
            try:
                # Check directory contents. If empty, it will raise StopIteration
                # and only in that case we delete the directory.
                next(menu_location.iterdir())
            except StopIteration:
                log.debug("Removing %s", self.start_menu_location)
                shutil.rmtree(self.start_menu_location, ignore_errors=True)
```

This reads a bit clearer to me. Let me know what you think.